### PR TITLE
CONCF-52 clean up globalNavigation related scripts

### PR DESCRIPTION
**Pull request by bognix**

### CODE REVIEW

**Comment by hakubo** [comment]
LGTM

File: extensions/wikia/GlobalNavigation/scripts/GlobalNavigationAccountNavigation.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	'use strict';
+	var $globalNavigation = $('#globalNavigation'),
+		$entryPoint = $('#AccountNavigation'),
+		loginAjaxForm = false;
+
+	/**
+	 * Handle click on entry point for logged in users.
+	 * Second click on entry point for logged in users is redirecting to user profile page.
+	 * This method should be removed after we unify the ux for anon and logged in.
+	 * @param event
```
Comment (vforge) [inline extensions/wikia/GlobalNavigation/scripts/GlobalNavigationAccountNavigation.js]: Please add @desc and ```@param {Event} event```

File: extensions/wikia/GlobalNavigation/scripts/GlobalNavigationNotifications.js [hunk 2]
```diff
@@ -33,89 +33,66 @@ require(
...
 				this.$notifications
-					.mouseenter( this.proxy( this.updateCounts ) )
-					.mouseenter( this.proxy( this.fetchForCurrentWiki ) );
+					.mouseenter(this.proxy(this.updateCounts))
+					.mouseenter(this.proxy(this.fetchForCurrentWiki));
 
-				this.$wallNotifications.add( $('#pt-wall-notifications') )
-					.on('click', '.notifications-markasread', this.proxy( this.markAllAsReadAllWikis ));
+				this.$wallNotifications.add($('#pt-wall-notifications'))
+					.on('click', '.notifications-markasread', this.proxy(this.markAllAsReadAllWikis));
```
Comment (hakubo) [inline extensions/wikia/GlobalNavigation/scripts/GlobalNavigationNotifications.js]: consider: `this.markAllAsReadAllWikis.bind(this);`

I am not changing it as I am not aware of all side effects

File: extensions/wikia/GlobalNavigation/scripts/GlobalNavigationDropdownsHandler.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+			dropdownOpts = registeredDropdowns[activeDropdown];
+			closeSelectedDropdown($dropdown, dropdownOpts, true);
+		}
+	}
+
+	/**
+	 * Close dropdown passed as argument.
+	 * If dropdown's onClose method returns false - this method's body is not executed.
+	 * This can be overridden by passing forceClose as true - it ignores onClose method return value.
+	 * @param jQueryObject $dropdown - jquery object representing dropdown
```
Comment (vforge) [inline extensions/wikia/GlobalNavigation/scripts/GlobalNavigationDropdownsHandler.js]: Please add ```@desc``` and change param format to ```@param {Event} event```

File: resources/wikia/modules/browserDetect.js [hunk 4]
```diff
@@ -50,6 +50,14 @@ define('wikia.browserDetect', ['wikia.window'], function (win) {
 		return !!userAgent.match(/iPad.+OS.[6,7].\d.+like.Mac.OS.+Safari/i);
 	}
 
+	function isTouchScreen() {
```
Comment (vforge) [inline resources/wikia/modules/browserDetect.js]: 1) please add documentation
2) can you browse through code and replace ``Wikia.isTouchScreen()``` with that? You could also remove former library / function.

File: resources/wikia/modules/delayedHover.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+	'use strict';
+
+	function attach(entryPoint, options) {
+		win.delayedHover(entryPoint, options);
+	}
+
+	return {
+		attach: attach
+	};
+});
```
Comment (vforge) [inline resources/wikia/modules/delayedHover.js]: Missing newline

File: resources/wikia/modules/menuAim.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+	'use strict';
+
+	function attach(entryPoint, options) {
+		win.menuAim(entryPoint, options);
+	}
+
+	return {
+		attach: attach
+	};
+});
```
Comment (vforge) [inline resources/wikia/modules/menuAim.js]: Missing newline

File: extensions/wikia/GlobalNavigation/scripts/GlobalNavigationNotifications.js (hunk 2 above)
Comment (bognix) [inline extensions/wikia/GlobalNavigation/scripts/GlobalNavigationNotifications.js]: Neither am I. But let's do so :+1:

Comment (bognix) [inline extensions/wikia/GlobalNavigation/scripts/GlobalNavigationNotifications.js]: After doing some reading I believe we should leave it as it is. Function.prototype.bind and $.proxy() are not exactly the same. Bind is always returning new function but $.proxy() returns new function only if proxy for this function and for given set of arguments doesn't exist. Notifications code is very fragile and we need to rewrite it anyway when new notification service would be created. Right now, the cost (QA) might be bigger than profit.

**Comment by vforge** [comment]
LGTM

**Pull request merged**